google/orbit
# Show capture length in the load capture widget

With this change, we add the capture length information to the load capture widget.

Now, when the user takes a new capture file or touches a previous capture file, the capture length information will be added to corresponding `CaptureFileInfo` and saved. (More use cases can be found in the following test cases).

Screenshots: [After the change](http://screenshot/47AFnYyrkQQgFVo); [Tooltips if the capture length is missing](http://screenshot/5qVQXAcjGGsREen)

Bug: http://b/198562119, http://b/227554942, http://b/227547320 

Test: 
1) Unit test; 
2) Run Orbit and take a new capture. Return to the connection window, see that the correct capture length value in the load capture widget; 
3) Run Orbit and on the connection window, pick a capture file with "--" in the "Capture length" column to load. Return to the connection window, see that the capture length value changed from "--" to the same value that we see in the capture main window timer label;
4) Rename a capture file from the capture main window. Return to the connection window, see that the renamed file also has the correct capture length value.
5) Close Orbit and modify a capture file from the file explorer. Run Orbit again and check that the capture length value changes to "--".

Besides, now we changed the `CaptureFileInfo` to only reflect the information of a capture file when it is changed inside Orbit. Take the file size as an example. Previously, `CaptureFileInfo` retrieves the file size from QFileInfo; while now, we save the file size in `CaptureFileInfo` and update the value only when we change / touch the file inside Orbit (`orbit_capture_file_info::Manager::AddOrTouchCaptureFile` should be called).
Then if the information from QFileInfo does not match the information from CaptureFileInfo, we say that the capture file might be modified outside Orbit and so the CaptureFileInfo is out of sync with the actual file.
